### ISSUE TITLE
[Discussion] Voice / Video call possibilities 

### ISSUE THREAD

**Issue by kskarthik** [r10s/messenger-android#70]
First, This app concept is amazing! 

Is it possible to provide voice / video call support ? 
Riot messenger which uses webRTC. Is it possible in the case of this app

Thank you :)

**Comment by r10s** [comment]
I have the rough idea, that this could be possible by peer-to-peer. 
To let the peers find each other, a normal mail could be used for "ringing", this mail would contain the IP-address of the caller then.
However, I'm no expert in this stuff, maybe there are other people around here who can implement it. Or who have more concrete ideas.

**Comment by kskarthik** [comment]
Any possibility to embed already existing WebRTC VoIP stack with android webview? https://github.com/jitsi/jitsi-meet

**Comment by r10s** [comment]
WebRTC sounds interesting, however, currently, I have higher priorities. Anyone else?

**Comment by KingDuckZ** [comment]
I came here looking for the same info. This app has a great concept, I think voice chat is the big feature that's still missing.

**Comment by r10s** [comment]
> voice chat

@KingDuckZ thank you for your post.
if "voice chat" means "voice messages": they are already available. goold old "telephone calls" are missing.

**Comment by KingDuckZ** [comment]
Yes I'm aware of that feature and it's a very welcome one!
I'm talking about real time telephone calls indeed. I know that would be a completely separate system from email messaging, but maybe once you send the caller's IP there's relatively simple approaches. Like using toxcore, or this webrtc thing or even some xmpp library.
If there was a bounty for this feature I'd be happy to chip some money in.

**Comment by asakura42** [comment]
Reopen this issue pls. Element (formerly Riot) has as I know Jitsi blobs and you can call through Element. Calling through delta may be simple and useful feature in our days.

https://eylenburg.github.io/im_comparison.htm wrote "However, the email protocol has its own limits, for example you can't conduct voice or video calls. Because of this, I don't think that Delta Chat will have a realistic chance of ever replacing WhatsApp." and that's right: we live in the epoch of speeches and voices.

I propose simple behavior: when Alice calls Bob, she sends email with start calling state. When Bob didn't answer and Alice drops phone, she sends second email to stop Bob's phone ringing. While delta works at background, it may work automatically.

**Comment by r10s** [comment]
> I propose simple behavior: when Alice calls Bob, she sends email with start calling state.

i think, meanwhile, it is pretty much like that:

there is the option to arrange video and audio calls through deltachat: eg. you can define a "video chat instance" in the Delta Chat  settings and invite all chat members to a room then - that can be the jitsi app then. 

big advantage is that this also works seemlessly with non-deta-clients. see https://delta.chat/en/2020-07-30-summer-update for some more details.

the feature is currently still in the "experimental" options, however quite stable and we are thinking about using announcing it as a regular feature.

**Comment by asakura42** [comment]
@r10s Yes. But users are expecting more natural way to call. Every messenger has a such feature, it's essential, I think. Now it's like "send invite and wait while Bob opens message and clicks link". Calls are fast and loud. You call and expect to listen 'hello'.

And it won't in any way interfere of the idea of communicating via email, on the contrary, it will help expand the audience of the application.

**Comment by r10s** [comment]
yip, things can be improved further, sure :) feel free to open a request at https://support.delta.chat (you can log in with your github handle there) (this issue tracker is for bugs and things someone subscribed to only)
